### DISCUSSION TITLE
Review of Zotero

### DISCUSSION THREAD

**Discussion by AlexeyMatveev686** [ONLYOFFICE/onlyoffice.github.io#244] [Polls]
<img src="https://raw.githubusercontent.com/ONLYOFFICE/onlyoffice.github.io/master/discussions-icons/zotero.svg">

**Comment by jbsbbruno** [comment]
The plugin is not working correctly, the moment the reference is inserted it stops.
Fedora 39 user linux kernel 6.5.12-300.fc39.x86_64. OnlyOffice version 7.5.1.23 (flatpak)

**Reply by VladimirProkofiev52** [reply]
Hello! @jbsbbruno 
I looked into your problem and couldn't reproduce it.
Please describe the problem in more detail.
Are you using the desktop version of the editor or the WEB version?

**Reply by jbsbbruno** [reply]
As the plugin already comes with onlyOffice, I didn't need to install it. I configured the plugin, associating the Zotero account normally. The problem occurs when creating a citation in the document, after selecting the citation and pressing the "insert citation" button, the citation is not inserted and the plugin keeps loading and no longer works.
![Captura de tela de 2023-11-28 11-20-38](https://github.com/ONLYOFFICE/onlyoffice.github.io/assets/152194985/df4aaf80-4d9a-48f4-94f8-93510dddd332)
![Captura de tela de 2023-11-28 11-20-56](https://github.com/ONLYOFFICE/onlyoffice.github.io/assets/152194985/40898ea1-0020-400c-ac81-a768af40be25)

**Reply by tvrbanec** [reply]
Same here. Debian 11.

**Reply by shibco** [reply]
I have this issue, and attempted to diagnose in #220 with no response from the plugin authors. Now my team has to complete our references manually. :sparkles:

**Reply by Misokados** [reply]
Hello,
I've checked desktop app v.9.2.1 and Zotero plugin v.1.0.6. All things are OK. Please provide us with detailed scenario to reproduce the situation.

**Comment by jbsbbruno** [comment]
The plugin still doesn't work! Even with the update to version 1.0.3 the plugin does not work.
The plugin was configured according to the instructions on the screen, it apparently works, it inserts the bibliography. However, when you try to insert a quote, the plugin does not work. It gets stuck on the loading screen.
![Captura de tela de 2023-12-29 09-14-25](https://github.com/ONLYOFFICE/onlyoffice.github.io/assets/152194985/62000b7a-94db-44f2-a252-23a36c4d9050)
![Captura de tela de 2023-12-29 09-14-56](https://github.com/ONLYOFFICE/onlyoffice.github.io/assets/152194985/9d56d5bc-721a-4c74-b464-34f12c62e42b)
![Captura de tela de 2023-12-29 09-15-08](https://github.com/ONLYOFFICE/onlyoffice.github.io/assets/152194985/d0664cc1-285c-4f12-b3c9-9f91da8be36d)
![Captura de tela de 2023-12-29 09-15-41](https://github.com/ONLYOFFICE/onlyoffice.github.io/assets/152194985/b672019e-fca5-40dc-9ba7-c1a0773e858e)

**Reply by jrom99** [reply]
Same here, they're tracking it on issue #220, but it works normally on the online version.

**Reply by Misokados** [reply]
Hello,
I've checked desktop app v.9.2.1 and Zotero plugin v.1.0.6. All things are OK. Please provide us with detailed scenario to reproduce the situation.

**Comment by geekyBoogiepop** [comment]
This issue still happening. Ubuntu 24.04 running OnlyOffice 8.0.1

**Reply by Misokados** [reply]
Hello,
I've checked desktop app v.9.2.1 and Zotero plugin v.1.0.6. All things are OK. Please provide us with detailed scenario to reproduce the situation.

**Comment by tvrbanec** [comment]
OnlyOffice developers are so close: full Zotero support and support for numbering pages with Roman numbers need to be included. Yet, it seems that they do not care. Their priority is PDF Editor. They do not realize that if we can use OnlyOffice in Academia, we will introduce it to students, and they will have the user base spreading.

**Reply by JAQuent** [reply]
I am using OnlyOffice for academic writing and the zotero plugin has some problems that make it really difficult to use.
1. Whenever I insert a citation, it jumps to a different part of the document and you have to scroll all the way back. When I already inserted a bibliography it usually jumps there. If not, it seem to jump to the first citation I inserted. This is obviously super frustrating.
2. It's impossible or unclear how to edit citations. For instance, I want to use APA style and change a citation from `(Bla et al., 2024)` to `Bla et al. (2024) suggested...` (or ` (see Bla et al, 2024)`) but as soon as the document refreshes these edits go away. This is also super frustrating.
3. Clicking on refresh by they way also jumps to the bibliography.

Overall, I think the zotero plugin is not really working well enough to be actually used professionally, which is a shame because OnlyOffice on its own is by far the best Office program for Linux that I have tested.  However, I can only use it with proper zotero support.

OS: Ubuntu 22.04
ONLYOFFICE Desktop Editors ver. 8.1.1.27

**Reply by JAQuent** [reply]
Also, one thing that seems to be missing is to convert the fields into simple text.

**Reply by jrom99** [reply]
> Also, one thing that seems to be missing is to convert the fields into simple text.

What would be the use case for this, though?

**Reply by JAQuent** [reply]
In my experience with words is that tracking changes and adding references doesn't really work well together so that's one but in general it would create a simpler version of the document that doesn't have a lot of fields.

**Reply by tvrbanec** [reply]
OnlyOffice developers just need to add the same functionalities that Zotero has in LibreOffice Writer and in Microsoft Word. No more, no less.

**Reply by jrom99** [reply]
> In my experience with words is that tracking changes and adding references doesn't really work well together so that's one but in general it would create a simpler version of the document that doesn't have a lot of fields.

It seems to me that it would be better to just fix the tracking part, wouldn't it?

As for the number of fields, are you talking about their appearance on the document, or about their existence itself? I've worked with docs with about 80-100 references and there seemed to be no problem due to the fields themselves.

> OnlyOffice developers just need to add the same functionalities that Zotero has in LibreOffice Writer and in Microsoft Word. No more, no less.

Ah, I agree now, removing and editing citations within the plugin itself is required for feature parity with [MS Word](https://www.zotero.org/support/kb/unlinking_citations) and [LibreOFfice](https://www.zotero.org/support/libreoffice_writer_plugin_usage)

Edit: the functionality to unlink all citations mentioned [here](https://www.onlyoffice.com/blog/2023/03/meet-updated-zotero-plugin) also wipes my Zotero API key, so it is a lot less useful than the ability to unlink specific citations in MS Word.

**Reply by Misokados** [reply]
Hello,
Please re-check the situation with the latest desktop app v.9.2.1 and Zotero plugin v.1.0.6. If the issue still persists, please post detailed scenario on https://community.onlyoffice.com/, we will check it out.

**Comment by kaustavg** [comment]
Unfortunately, the Zotero plugin isn't fully supported by OnlyOffice. For example, new references added to a document does not renumber old references [Issue #336](https://github.com/ONLYOFFICE/onlyoffice.github.io/issues/336).

Other issues as well: https://github.com/ONLYOFFICE/onlyoffice.github.io/issues/220 https://github.com/ONLYOFFICE/onlyoffice.github.io/issues/368

**Reply by Misokados** [reply]
Hello,
the mentioned issue (bug #67682) has been fixed in the plugin v.1.0.6. Please re-check the situation.
As for the other issue, the workaround solution has been provided here: https://github.com/ONLYOFFICE/onlyoffice.github.io/issues/220#issuecomment-3696613774
If the issue still persists, please post it on https://community.onlyoffice.com/ and provide us with step-by-step scenario.

**Reply by tvrbanec** [reply]
It is still not working as it should. Please, read my post.
<img width="900" height="623" alt="Zotero_OO_plugin_new" src="https://github.com/user-attachments/assets/1b14f612-9f0f-4c01-828a-6d780cee0718" />
<img width="670" height="248" alt="Zotero_browser_plugin" src="https://github.com/user-attachments/assets/9c5a1975-efbd-4ad4-8e81-3c3aa9a4aa64" />
<img width="563" height="281" alt="Zotero_standalone" src="https://github.com/user-attachments/assets/14d55ddd-1b6d-4fb5-b69d-120a6309c95b" />

**Comment by Visel128** [comment]
This plugin has one main problem. Official Zotero plugins for LibreOffice and MS Office fetch all information directly from Zotero application that currently running  on your computer. However this plugin fetch all data from your Zotero account in the web. Because of this that plugin requires continuous connection to the internet and requires the API Key unlike the official Zotero plugins that not require this.
Additionally this plugin don't compatible with the official Zotero plugins therefore to create documents in MS or Libre and in Only simultaneously it is impossible.

**Reply by jrom99** [reply]
Just to be sure, are LibreOffice and MS Word citations made with the official Zotero plugin compatible with each other? Does this extends to the official Zotero plugin for Google Docs?

That said, I personally don't mind that the plugin uses online data instead of the standalone app to fetch and update citations (it could use a local cache though), since making it properly communicate with the Zotero standalone app would be really hard for sandboxed versions of OnlyOffice (snap, flatpak, online).

The current implementation for LibreOffice (https://github.com/zotero/zotero-libreoffice-integration) for example has to start a local service to communicate indirectly with the app, and I'm not sure how the MS Word (https://github.com/zotero/zotero-word-for-windows-integration) does it, since it depends on a multiple `.dll` files, somehow including both Firefox and the Zotero standalone app. I didn't take a good look at the code in the Google Docs implementation (https://github.com/zotero/zotero-google-docs-integration), so I'm not sure whether it communicates with the standalone app at all.

As for the interoperability, going by the source code for the LibreOffice implementation:

https://github.com/zotero/zotero-libreoffice-integration/blob/2183efa32284c061883628029ce030aaba7ec63e/build/source/org/zotero/integration/ooo/comp/Document.java#L94-L101

I'm fairly sure that compatibility with LibreOffice is fairly achievable as long as the plugin in OnlyOffice starts using the same prefixes and suffixes in the citation field, since both seem to use the same CSL notation to store its data (this WILL require a breaking change in OnlyOffice Zotero's plugin though). But I would have to take a deep look at the code (or compare example documents to check if this is indeed the case).

**Reply by Misokados** [reply]
Hello,
I'm not sure that I understood the idea completely. Do you mean the scenario that user installs Zotero connector locally and the plugin syncs with it instead of syncing with account in the web? https://www.zotero.org/download/

**Reply by Visel128** [reply]
> Hello, I'm not sure that I understood the idea completely. Do you mean the scenario that user installs Zotero connector locally and the plugin syncs with it instead of syncing with account in the web? https://www.zotero.org/download/

Hello, not exactly! I mean that a user can install the Zotero desktop application and a plugin for your document editor. After that, the Zotero plugin fetches all bibliographic data from the Zotero desktop application locally.
More visually in this video: https://youtu.be/WcIa7iFXnLs

Personally, I really like this approach because it allows you not to depend on API keys and an Internet connection.

**Reply by Misokados** [reply]
It seems that the feature for working with local Zotero is already available::
<img width="893" height="579" alt="image" src="https://github.com/user-attachments/assets/4bd67e46-d238-4386-aead-81f0f144cfa5" />
Please clarify if this is what you're looking for.

**Reply by Visel128** [reply]
Yes, I just tested it on my Windows and Linux systems. It really works. Apparently, this functionality of the plugin appeared recently and I did not have time to track it. Thank you so much for the clarification!

**Comment by tvrbanec** [comment]
To my surprise, it is actually working now, as I can check.
Some problems that should be fixed:
1. Setting referencing style is painful (very bad search). 
2. There is no editor for style customisation.
3. Bibliography is unspeakably bad: some extra spaces on the begining of every item, two extra line breaks after items in some styles, extra line breaks randomly **in** items.

**Reply by jrom99** [reply]
>Bibliography is unspeakably bad: some extra spaces on the begining of every item, two extra line breaks after items in some styles, extra line breaks randomly in items.

Although I can confirm that the extra spaces and two line breaks after items happen, I have never dealt with the extra line breaks inside an item, is it not something in the item itself in Zotero?

**Reply by Misokados** [reply]
We’re continuing to work on improving the plugin, but we need your assistance with this matter. Please record a video showing your actions, and we will check it out.

**Reply by tvrbanec** [reply]
I will, in a day or two, three. I'm unusually busy. Please be patient.

**Reply by tvrbanec** [reply]
Finally, I made it.
1st: 1) & 2) is to be confirmed.
2nd: Unfortunately, now I cannot see the extra lines I was complaining about before. :(
3rd: I typed in the document (in the video) what I saw as not implemented, or poorly implemented.

https://github.com/user-attachments/assets/23dcc691-5a76-45c8-b060-9d7e2df7e2ba

**Reply by Misokados** [reply]
Thank you for the video file! Please clarify the desktop app and the plugin versions. I've tried to reproduce the described situations on the latest versions, but it seems that some of them are OK (I see numbering, aligned to the left, no double spacing)
<img width="2174" height="1182" alt="image" src="https://github.com/user-attachments/assets/61d7de25-18ac-4756-8230-203764a049e8" />

**Reply by Misokados** [reply]
Dear @tvrbanec 
We're checking the situation. I will contact you as soon as possible.

**Reply by tvrbanec** [reply]
Thanks. It is a Debian OS, not Windows...

**Reply by tvrbanec** [reply]
I don't know why is generated Bibliography so ugly and it is not sorted. APA7 (no ampersand).
<img width="976" height="712" alt="Screenshot_20260301_224738" src="https://github.com/user-attachments/assets/dec03bef-7b62-448f-bae9-863f69e1d31b" />

**Reply by Misokados** [reply]
Dear @tvrbanec 
Could you please double-check the Zotero plugin version? It seems that you're using the outdated version, the current one is 1.0.6.
If so, please update the plugin and re-check the situations

**Reply by tvrbanec** [reply]
As I mentioned previously in https://github.com/ONLYOFFICE/onlyoffice.github.io/discussions/244#discussioncomment-15870159 I already use 1.0.6.

**Reply by Misokados** [reply]
Thank you! I had to confirm that because the provided video file was recorded in the older version of the plugin. Could you please confirm whether the described issues are still present in v1.0.6?
1. No ability to edit citations (time code 00:15).
2. Missing numbering when inserting multiple bibliography entries (01:29).
3. Double space at the beginning of the insertion (02:10).
4. It is unclear why a line break using Shift+Enter is used instead of Enter when inserting the bibliography (02:30).
5. When inserting multiple lines, the last line is aligned to the right instead of to the left (04:00).

**Reply by tvrbanec** [reply]
> Thank you! I had to confirm that because the provided video file was recorded in the older version of the plugin. 
It is not in older version, I told you that before.

Could you please confirm whether the described issues are still present in v1.0.6?
> 
> 1. No ability to edit citations (time code 00:15).

Still not fixed, one cannot edit ones inserted citations, only if delete it first.

> 2. Missing numbering when inserting multiple bibliography entries (01:29).

No numbering.

> 3. Double space at the beginning of the insertion (02:10).

Stil double space.

> 4. It is unclear why a line break using Shift+Enter is used instead of Enter when inserting the bibliography (02:30).

Stil Shift + Enter.

> 5. When inserting multiple lines, the last line is aligned to the right instead of to the left (04:00).

That is OK now, althought I would prefer justify exept for last line of reference.

**Reply by Misokados** [reply]
Thank you, we're checking the mentioned situations. I will contact you shortly.

**Reply by Misokados** [reply]
Dear @tvrbanec 
We have released plugin version 1.0.7. Please re-check your reports using the latest version.
If it's possible, please record a new video demonstrating with the updated plugin.

**Comment by Visel128** [comment]
I was working some time in the OnlyOffice with Zotero plugin with local Zotero and found some errors and shortcomings.
1. Adding each new citation, when the bibliography is created, leads to an unexpected camera switch to this citation in the list of bibliography. This fact is very, very inconvenient.
2. It seems that in this plugin there is no way to specify specific pages in the source, as, for example, in the Word plugin. This circumstance greatly limits the functionality and scope of application.
I am attaching a video with an illustration.

https://github.com/user-attachments/assets/6773c5a8-76c3-4772-840a-740426dc0cd4

**Reply by Misokados** [reply]
Hello @Visel128 
We have released the Desktop app v.9.3 yesterday, please update the app and re-check the situation. 
Also please confirm that you're using the latest Zotero plugin version (v.1.0.6).

**Reply by Visel128** [reply]
Yes, I just checked. There is no bug from point 1. Everything is working fine with the new version of the app. Thank you so much for your responsiveness!

**Reply by Visel128** [reply]
Please tell me, are there any plans to add the function of specifying specific pages when adding a citation?

**Reply by tvrbanec** [reply]
> Please tell me, are there any plans to add the function of specifying specific pages when adding a citation?

It is already added.

**Reply by Visel128** [reply]
> > Please tell me, are there any plans to add the function of specifying specific pages when adding a citation?
> 
> It is already added.

Okey, in this way please explain me how to use this function. I don't found it. Just in case, I'm referring to this function: 
https://youtu.be/MmR4VJnSJC0

**Reply by tvrbanec** [reply]
When you select a source, you do it by clicking the checkbox on the left side. You also have a drop-down menu from the right side. There it is.

**Reply by Visel128** [reply]
> When you select a source, you do it by clicking the checkbox on the left side. You also have a drop-down menu from the right side. There it is.

Yes, I found it. Thank you very much!

**Comment by zoroaster1x** [comment]
If anyone is interested, I made my own version of a Zotero Integration plugin which should be more similar to the Libreoffice version; https://github.com/zoroaster1x/ZoteroPlus

**Reply by Misokados** [reply]
That’s awesome! We’re always happy to see the community finding their own ways to use the product.

**Reply by tvrbanec** [reply]
Great! Thank you very, very much!!!

**Comment by laurareddybythatagari-pixel** [comment]
> ![](https://raw.githubusercontent.com/ONLYOFFICE/onlyoffice.github.io/master/discussions-icons/zotero.svg)

This is getting better, was FINALLY able to get some real academic work done in it, still needs a bit of work but its finally usable, thank you

**Comment by langdine** [comment]
It’s not working—just stuck on loading in local connection (May 19, 2026). 
Info: zotero 9.0.3 .
Perhaps the communication protocol has changed?
<img width="1202" height="779" alt="{0C2614A8-C07E-4474-B5E6-C38DFA087BD4}" src="https://github.com/user-attachments/assets/d1a8874d-d425-43fd-bb55-0b6079e0eec1" />

**Reply by Misokados** [reply]
Hello @langdine 
I've just checked it on the desktop app v.9.4, plugin v.1.0.7 All things are OK.
Could you please re-check the situation and confirm whether it still occurs?
